packages/components/src/components/editor/condition-builder/ConditionGroup.tsx

```diff
@@ -0,0 +1,66 @@
+import { IvyIcons } from '@axonivy/ui-icons';
+import { Condition, type ConditionProps } from './Condition';
+import { useConditionBuilderContext, type LogicalOperator } from './ConditionBuilderContext';
+import { BasicSelect } from '@/components/common/select/select';
+import { Flex } from '@/components/common/flex/flex';
+import { Label } from '@radix-ui/react-label';
+import { Button } from '@/components/common/button/button';
+
+export interface ConditionGroup {
+  conditions: Condition[];
+  logicalOperator: LogicalOperator;
+}
+
+interface ConditionGroupProps extends Pick<ConditionProps, 'groupIndex'> {
+  group: ConditionGroup;
+  groupCount: number;
+}
+
+export const ConditionGroup = ({ group, groupIndex, groupCount }: ConditionGroupProps) => {
+  const { updateLogicalOperator, addCondition, removeConditionGroup, conditionMode, logicalOperatorOptions } = useConditionBuilderContext();
+
+  return (
+    <Flex direction='column' gap={2} className='condition-builder__group'>
+      <Flex
+        direction='column'
+        style={
+          conditionMode === 'nested-condition'
+            ? { border: 'var(--basic-border', borderRadius: 'var(--border-r2)', padding: 'var(--size-2)' }
+            : undefined
+        }
+        gap={2}
+      >
+        {conditionMode === 'nested-condition' && (
+          <Flex direction='row' justifyContent='space-between'>
+            <Label>{`Group ${groupIndex + 1}`}</Label>
+            <Button onClick={() => removeConditionGroup(groupIndex)} icon={IvyIcons.Trash} aria-label='Remove Group' />
+          </Flex>
+        )}
+        {group.conditions.map((condition, conditionIndex) => (
+          <Condition
+            key={conditionIndex}
+            condition={condition}
+            conditionIndex={conditionIndex}
+            groupIndex={groupIndex}
+            conditionsCount={group.conditions.length}
+          />
+        ))}
+        <Button onClick={() => addCondition(groupIndex)} icon={IvyIcons.Plus} aria-label='Add Condition' variant='outline'>
+          Add Condition
+        </Button>
+      </Flex>
+      {groupIndex < groupCount - 1 && conditionMode === 'nested-condition' && (
+        <BasicSelect
+          items={Object.entries(logicalOperatorOptions)
+            .slice(0, -1)
```

Comment:
remove empty one?

Reply:
No for some reason if i do it with Object.entries() the last array is always "parameters" no idea why, so i remove it.
![grafik](https://github.com/user-attachments/assets/a117a589-e1e1-4bd8-a000-342eef3441df)

Reply:
seems like a hack to me. we should better know why this is and fix it at the source :)

Reply:
Yes I agree 😅 However, for some reason after I moved the hole Object.entries().map logic into the ConditionBuilderContext, there was no longer the issue with this parameters and i was able to remove the slice 👍